fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/AbstractLoanRepaymentScheduleTransactionProcessor.java

```diff
@@ -546,12 +547,21 @@ private List<LoanTransaction> getMergedTransactionList(List<LoanTransaction> tra
     protected void createNewTransaction(LoanTransaction loanTransaction, LoanTransaction newLoanTransaction,
             ChangedTransactionDetail changedTransactionDetail) {
         loanChargeValidator.validateRepaymentTypeTransactionNotBeforeAChargeRefund(loanTransaction.getLoan(), loanTransaction, "reversed");
+
+        // Save external ID before clearing it to check if this was a user-initiated transaction
+        ExternalId originalExternalId = loanTransaction.getExternalId();
+
         loanTransaction.reverse();
         loanTransaction.updateExternalId(null);
         newLoanTransaction.copyLoanTransactionRelations(loanTransaction.getLoanTransactionRelations());
-        // Adding Replayed relation from newly created transaction to reversed transaction
-        newLoanTransaction.getLoanTransactionRelations().add(
-                LoanTransactionRelation.linkToTransaction(newLoanTransaction, loanTransaction, LoanTransactionRelationTypeEnum.REPLAYED));
+
+        // Create REPLAYED relation for user-initiated transactions (those with external IDs)
+        // This distinguishes legitimate business operations from internal processing artifacts
+        if (originalExternalId != null) {
```

Comment:
external id has not much to do whether it is business operation or internal processing... not necessarily enabled external id for loan transactions...